### PR TITLE
Add REST core and refactor dependencies

### PR DESCRIPTION
**Pull request by gogit**
REST core module  and refactored dependencies to raml-parser

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,13 +8,21 @@ on [Keep a CHANGELOG](http://keepachangelog.com/). This project adheres to
...
 the `aggregate-snapshot-service` module. Requires the `aggregate-snapshot-repository-liquibase`
 schema to be applied to the event store.
 - JSON schema matching support in the JsonEnvelopeMatcher
+- Added new module 'rest-core'
+- ParameterType and ActionType for HTTP raml conversion
+- Added 'rest-client-core' as a standard dependency for event-processor'
+- Added 'rest-core' as a standard dependency for 'generators'
+- Added 'rest-core' as a standard dependency for 'rest-adapter-core'
+- Added 'rest-core' as a standard dependency for 'rest-client-generator'
+- Added 'rest-core' as a standard dependency for 'rest-client-core'
```
Comment (purple52) [inline CHANGELOG.md]: All the comments here for est-core are too low-level. I think there should be a single entry under Changed that explains the overall change, and what it achieves.

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,13 +8,21 @@ on [Keep a CHANGELOG](http://keepachangelog.com/). This project adheres to
...
+- Added 'hamcrest-date' as a standard dependency for 'test-utils-core'
```
Comment (purple52) [inline CHANGELOG.md]: This one is okay.

File: CHANGELOG.md [hunk 3]
```diff
@@ -24,6 +32,10 @@ schema to be applied to the event store.
 - RestProcessor now allows metadata from payload to be merged with headers instead of being
 overwritten
 
+### Removed
+- 'raml-parser' as a dependency for 'rest-adapter-core'
+- 'rest-adapter-core' as a dependency for 'rest-client-core'
```
Comment (purple52) [inline CHANGELOG.md]: These are part of the "change" as per the earlier comment.

**Comment by gogit** [comment]
Changelog changes are complete

File: CHANGELOG.md (hunk 1 above)
Comment (gogit) [inline CHANGELOG.md]: done

File: CHANGELOG.md (hunk 2 above)
Comment (gogit) [inline CHANGELOG.md]: same as before

File: CHANGELOG.md (hunk 3 above)
Comment (gogit) [inline CHANGELOG.md]: done

**Pull request merged**